### PR TITLE
Add secrets list and delete action

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HAC-4557
https://issues.redhat.com/browse/HAC-4559

## Description

This PR adds Secret list page and delete action on the list.

Additional details:

-  `Learn more` button will be added once docs team provides the required `id`.
- `Add Secret` button will be added in separate PR while we implement Add secret feature.

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

Access Denied Page:

AccessDenied

Secret List page:

SecretList

Row with labels:

Screenshot 2023-08-22 at 12 44 02 PM

Delete Secret action:

https://github.com/openshift/hac-dev/assets/9964343/73faff0d-5650-4073-9871-8f673eaaa421

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc: @pankajshivpuje @Misjohns

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
@pankajshivpuje  I will add `Add secret` button later in separate PR once add secret form is ready and `learn more` link will be added when docs team provides the id.

Could you please provide the microcopy for the secrets empty state?

image

File: src/types/secret.ts [hunk 1]
```diff
@@ -19,24 +31,115 @@ type linkServiceAccount = {
...
+  Error = 'Error',
+  Unknown = '-',
+}
+
+export enum RemoteSecretStatusType {
+  Deployed = 'Deployed',
+  DataObtained = 'DataObtained',
+}
+export interface Condition {
+  lastTransitionTime: string;
```
Comment (rottencandy) [inline src/types/secret.ts:69]: Looks like this is causing build failure because there is already another export called `Condition`

Comment (karthikjeeyar) [reply src/types/secret.ts]: fixed it.

**Review by rottencandy — APPROVED** [review]
/lgtm

**Pull request merged**